pkg/controller/admissionchecks/multikueue/workload.go

```diff
@@ -61,6 +63,12 @@ type jobAdapter interface {
 	CopyStatusRemoteObject(ctx context.Context, localClient client.Client, remoteClient client.Client, key types.NamespacedName) error
 	// Deletes the Job in the worker cluster.
 	DeleteRemoteObject(ctx context.Context, remoteClient client.Client, key types.NamespacedName) error
+	// KeepAdmissionCheckPending returns true if the state of the multikueue admission check should be
+	// kept Pending while the job runs in a worker. This might be needed to keep the managers job
+	// suspended and not start the execution locally.
+	// Going forward, if the job management is controlled at object label, this should get a workload
+	// reference as argument.
+	KeepAdmissionCheckPending() bool
```

Comment:
Ideally, I would like this not to exist, we could disable the job controller in the e2e tests, and I prefer to have the production code simple, rather than testing. Analogously to what we do for JobSet by not installing the CRD.

Can be a follow up.

Reply:
I still don't understand still. 

Are you talking about role sharing, so that management cluster is also a worker? 

If so, we are going to create a pair of jobs, and mirror within the cluster as if there was another cluster, for what we discussed so far, but the details are not laid out in kęp yet. 

Since it is not confirmed yet I prefer not making this statements about the future

Reply:
Let's keep it for now, since jobset has a different behavior than batchjob.

Reply:
thanks